synapse/cores/common.py

```diff
@@ -2832,6 +2836,76 @@ def snapTufosByDark(self, name, valu=None, mintime=None, maxtime=None, limit=Non
         idens = list(set([r[0][::-1] for r in rows]))  # Unique the idens we pull.
         return self._initTufoSnap(idens)
 
+    def _getBlobValu(self, key, default):  # pragma: no cover
+        raise s_common.NoSuchImpl(name='_getBlobValu', mesg='Core does not implement _getBlobValu')
+
+    def _setBlobValu(self, key, valu):  # pragma: no cover
+        raise s_common.NoSuchImpl(name='_setBlobValu', mesg='Core does not implement _setBlobValu')
+
+    def _getCoreType(self):  # pragma: no cover
+        raise s_common.NoSuchImpl(name='getCoreType', mesg='Core does not implement getCoreType')
+
+    def getCoreType(self):
+        return self._getCoreType()
+
+    # TODO: Wrap this in a userauth layer
+    def getBlobValu(self, key, default=s_common.novalu):
+        '''
+        Get a value from the blob key/value (KV) store.
+
+        This resides below the tufo storage layer and is Cortex implementation
+        dependent. In purely memory backed cortexes, this KV store may not be
+        persistent, even if the tufo-layer is persistent, through something
+        such as the savefile mechanism.
+
+        Notes:
+            Data which is retrieved from the KV store is msgpacked, so caveats
+            with that apply.
+
+        Args:
+            key (str): Value to retrieve
+            default: Optional Value to return if the key is not present in the
+                     KV store.  By convention, the Cortex implementation should
+                     not set the requested value to the default upon a missing
+                     key, as this mechanism may be used to provide for
+                     bootstrapping changes at the storage layer of a Cortex.
+
+        Returns:
+            The value from the KV store or the default valu if set.
+
+        Raises:
+            NoSuchImpl: If the storage layer does not implement the blob KV
+                        store.
+            NoSuchName: If the KV store does not contain the requested key and
```

Comment:
This causes the need for callers to try/catch for NoSuchName and gets a bit ugly for the standard use case.  Lets mod this to return None when not found (which would also make it more consistent with the other API conventions